src/coreclr/vm/amd64/AsmHelpers.asm

```diff
@@ -447,6 +447,25 @@ NESTED_ENTRY OnCallCountThresholdReachedStub, _TEXT
         TAILJMP_RAX
 NESTED_END OnCallCountThresholdReachedStub, _TEXT
 
+extern JIT_PatchpointWorkerWorkerWithPolicy:proc
+
+NESTED_ENTRY JIT_Patchpoint, _TEXT
+        PROLOG_WITH_TRANSITION_BLOCK
+
+        lea     rcx, [rsp + __PWTB_TransitionBlock] ; TransitionBlock *
+        call    JIT_PatchpointWorkerWorkerWithPolicy
+
+        EPILOG_WITH_TRANSITION_BLOCK_RETURN
+        TAILJMP_RAX
+NESTED_END JIT_Patchpoint, _TEXT
+
+; first arg register holds iloffset, which needs to be moved to the second register, and the first register filled with NULL
+LEAF_ENTRY JIT_PartialCompilationPatchpoint, _TEXT
```

Comment:
Looks like we can remove  the need of separate`JIT_PartialCompilationPatchpoint` if we update its only usage:

https://github.com/dotnet/runtime/blob/d8a611799782afc0b14c214c8b541a0c3f6d3663/src/coreclr/jit/patchpoint.cpp#L241-L242

to be:
```c++
GenTree* nullCounter = compiler->gtNewIconNode(0, TYP_I_IMPL);
GenTreeCall* helperCall =
    compiler->gtNewHelperCallNode(CORINFO_HELP_PATCHPOINT, TYP_VOID, nullCounter, ilOffsetNode);
```